File: TreeViewAdvDemo/C#/Form1.cs
using Syncfusion.Windows.Forms;
using System;
using System.Windows.Forms;

namespace TreeViewAdvNodeId
{
    public partial class Form1 : MetroForm
	{
        #region Constructor
        public Form1()
		{
			InitializeComponent();
            for (int nodeId = 0; nodeId <= 10000; nodeId++)
			{
				//Custom node for ID propety
				CustomTreeNodeAdv customNode = new CustomTreeNodeAdv();
                customNode.ID = nodeId;
                customNode.Text = "Node" + nodeId.ToString();
                this.treeViewAdv1.Nodes.Add(customNode);
			}
            this.buttonAdv1.Click += buttonAdv1_Click;
		}
        #endregion
        
        void buttonAdv1_Click(object sender, EventArgs e)
        {
            //Iterating the nodes in the TreeViewAdv
            foreach (CustomTreeNodeAdv item in this.treeViewAdv1.Nodes)
            {
                //To get the TextBox value
                string textvalue = item.ID.ToString();
                if (this.integerTextBox1.Text == textvalue)
                {
                    //Get the node by its ID
                    MessageBox.Show(item.Text);
                }
            }
        }
	}
}

File: TreeViewAdvDemo/VB/Form1.vb
Imports Syncfusion.Windows.Forms
Imports System
Imports System.Windows.Forms

Namespace TreeViewAdvNodeId
    Partial Public Class Form1
        Inherits MetroForm
#Region "Constructor"
        Public Sub New()
            InitializeComponent()
            For nodeId As Integer = 0 To 10000
                'Custom node for ID propety
                Dim customNode As New CustomTreeNodeAdv()
                customNode.ID = nodeId
                customNode.Text = "Node" & nodeId.ToString()
                Me.treeViewAdv1.Nodes.Add(customNode)
            Next nodeId
            AddHandler Me.buttonAdv1.Click, AddressOf buttonAdv1_Click
        End Sub
#End Region

        Private Sub buttonAdv1_Click(ByVal sender As Object, ByVal e As EventArgs)
            'Iterating the nodes in the TreeViewAdv
            For Each item As CustomTreeNodeAdv In Me.treeViewAdv1.Nodes
                'To get the TextBox value
                Dim textvalue As String = item.ID.ToString()
                If Me.integerTextBox1.Text = textvalue Then
                    'Get the node by its ID
                    MessageBox.Show(item.Text)
                End If
            Next item
        End Sub
    End Class
End Namespace